python/mlx/nn/losses.py

```diff
@@ -372,3 +372,45 @@ def log_cosh_loss(
     loss = mx.logaddexp(errors, -errors) - math.log(2)
 
     return _reduce(loss, reduction)
+
+
+def cosine_similarity_loss(
+    inputs: mx.array,
+    targets: mx.array,
+    eps: float = 1e-8,
+    reduction: str = "none",
+) -> mx.array:
+    r"""
+    Computes the Cosine Similarity loss between inputs and targets.
+
+    Note that inputs must be between -1 and 1. When it is a negative number
+    between -1 and 0, 0 indicates orthogonality and values closer to -1 indicate
+    greater similarity. The values closer to 1 indicate greater dissimilarity.
+    This makes it usable as a loss function in a setting where you try to
+    maximize the proximity between predictions and targets. If either
+    `inputs` or `targets` is a zero vector, cosine similarity will be 0
+    regardless of the proximity between predictions and targets.
```

Comment:
Why do the inputs need to be between `-1` and `1`? I think you can remove this paragraph.